### ISSUE TITLE
Consider using WebP instead of PNG for image assets

### ISSUE THREAD

**Issue by pepoluan** [endless-sky/endless-sky#10064]
### Problem Description

Currently, all the PNG files inside the `images` directory takes more than 100 MiB, 178'830'063 bytes, to be exact:

```
$ { find . -type f -name "*.png" -printf "%s+"; echo 0; } | bc
178830063
```

Before the advent of more modern image formats, it is understandable that the image format chosen was PNG, because JPEG does not support the necessary alpha channel.

However, with WebP maturing and JPEG XL increasing in popularity, we should revisit this decision.

For example, after a quick-n-dirty conversion from PNG to WebP using the following command:

```
$ while read src; do cwebp -q 80 "$src" -o "${src%.*}.webp"; done < <(find . -type f -name "*.png")
```

I ended up with only _one-sixth_ the original size; 29'504'104 bytes, to be exact:

```
$ { find . -type f -name "*.webp" -printf "%s+"; echo 0; } | bc
29504104
```

I believe a saving of nearly 150 MB is quite significant.

### Related Issue Links

* Somewhat related to #4980 , though that one is more about animation frames.
* Strongly related: #9880 

### Desired Solution

* Convert all PNG images to WebP images
* Use `libwebp` to decode the new image format into in-memory bitmap for use

### Alternative Approaches

There is one drawback of using lossy compression for storage: **Generational Loss**

Successive editing and saving will slowly degrade the images.

To prevent this, we can use this strategy:

1. Move the PNG-format images to a different repo.
2. Changes to the images should be done _only_ in this repo.
3. Regularly (on change) generate a whole set of WebP images from these files as "release"
4. Grab the "release" images and use them in the `images` directory

**Comment by danaris** [comment]
I don't know offhand the differences between the formats, but HEIC is another modern image codec designed to be more space-efficient. I would suggest it also be considered, and compared, before deciding on a format.

**Comment by eebop** [comment]
Heic is proprietary to apple and requires a paid subscription on non-mac machines

**Comment by danaris** [comment]
So far as I can tell, it is not proprietary—it is an ISO standard, and can be implemented by anyone. According to https://en.wikipedia.org/wiki/High_Efficiency_Image_File_Format it is fully supported by Windows 11 and Android—it requires a paid purchase for Windows 10 to support specifically the ones that Apple creates by default (HEVC encoding), but not the format as a whole. It's not clear from this whether it's a one-time purchase or a subscription.

Do you have a source that contradicts this information?

**Comment by Zitchas** [comment]
To be honest, I detest dealing with webp stuff. Not everything opens it, and everytime I get an image in WebP it just means extra work. I convert everything *out* of WebP at first opportunity.

So aside from windows 10 needing a paid subscription (which I'm not paying, thank you very much.), how about those on variations of Linux and older Windows? Currently, our only graphical requirement is "Runs OpenGL 3.0 or greater". I don't think the gains are worth changing that to "Runs OpenGL 3.0 and has WebP support".

**Comment by RisingLeaf** [comment]
This is just premature optimisation, as Zitchas said webp is not as widespread as png and jpeg.

If we really wanted to save space and time we could think about saving precompiled pictures. But that's for later down the road.

I am also against adding webp support as that would be another dependency of the project (we already have a lot).

**Comment by Zitchas** [comment]
> I feel the discussion slightly beside the point: User or developer platform is irrelevant. You're ultimately preparing some storage into textures ready for the render pipeline - so it depends entirely on the library used to load those textures, including licensing.

A couple things here.

a) As a project with no formal organization, there's no entity to sign or pay for a licensing agreement.

b) One of the core tenets of the project has been that end-users can add their own graphics and data as easily as pasting a random image from their picture folder into the correct location, along with <5 mins of editing with their preferred text editor. I am aware of a number of contributors, myself included, who have contributed both code and content with this level of software.

c) As an aspect of B, we've made something of an effort to ensure that all the content (visual and otherwise) remains easily accessible to everyone in order to encourage community participation and contribution. 

d) Low appetite for putting work into adopting things that are hostile to FOSS.

**Comment by derei** [comment]
I agree that webp seems like a pain in the rear, and most people (including myself) are tempted to avoid it, and it only causes frustration especially when downloading a webp image from a website. However, webp has its own undeniable advantages if used  mindfully. And it will only grow and become widely accepted because of that. So, whilst I get the reluctance and resistance in considering it, perhaps it's worth thinking about:

Reasons why Webp should still be considered eventually (at some point down the timeline):
- significantly smaller file sizes (25-30% compared to png)
- supports alpha and animation
- is FOSS [(BSD License 3)](https://en.m.wikipedia.org/wiki/WebP). Yes, it was developed by Google, but it has been released as FOSS.

And for simplifying user experience, perhaps an internal asset converter or an online conversion tool, could be made. So the average user still can use their png file without needing advanced software. The point of the in-game converter would be to take the user-submitted png, convert it, and then automatically delete the original png.

**Comment by VaelophisNyx** [comment]
Supporting WebP seems like a ton of work for not much payoff honestly? If you want high quality animations in compact formats, AVIF and APNG exist for that (APNG even supports standard PNG handling as well). If you want a compact image format, PNG can also do that well enough by just turning up the compression a bit more. More libraries needed, and more fussy work to develop around for plugins and new content with converting to/from it.
